articles/frameworks/vue.md

```diff
@@ -1,3 +1,104 @@
+1. Select **Empty**, click OK.
+
+##### Configure the project startup:
+1. Open the file **./Startup.cs**, and add the following lines to the Configure method:
+
+```c#
+app.UseDefaultFiles() // Enables default file mapping on the web root.
+app.UseStaticFiles(); // Marks files on the web root as servable.
+```
+
+### Install the vue-cli
+
+To install the vue-cli, open a Command Prompt and type `npm install --g vue-cli` or `npm install -g @vue/cli` for version 3.0 (currently in beta).
+
+#### Use the vue-cli to scaffold a new client application:
+1. Go to your Command Prompt and change directory to your project root folder.
+1. Type `vue init webpack ClientApp` and follow the additional questions.
+
+##### Tweak the webpack config to output the built files to wwwroot:
+1. Open the file **./ClientApp/config/index.js**, and change the `build.index` and `build.assetsRoot` to wwwroot path:
+
+```js
+// Template for index.html
+index: path.resolve(__dirname, '../../wwwroot/index.html'),
+
+// Paths
+assetsRoot: path.resolve(__dirname, '../../wwwroot'),
+```
+
+##### Indicate the project to build the ClientApp everytime building is triggered:
+1. Go to **Project properties > Build Events**
+1. On **Pre-build event command line** type `npm --prefix ./ClientApp run build`.
+
+##### Configure webpack's output module names:
+1. Open the file **./ClientApp/build/webpack.base.conf.js**, and add the following properties to the output property:
+
+```js
+devtoolModuleFilenameTemplate: '[absolute-resource-path]',
+devtoolFallbackModuleFilenameTemplate: '[absolute-resource-path]?[hash]'
+```
+
+#### Use the vue-cli 3.0 (currently in beta) with TypeScript
+1. Go to your Command Prompt and change directory to your project root folder.
+1. Type `vue create ClientApp` and choose **Manually select features**.
+1. Choose Typescript and any other options desired.
+1. Follow the additional questions.
+
+##### Configure Vue.js-TypeScript project
+1. Open the file **./ClientApp/tsconfig.json** and add to the compiler options `noEmit:true` for avoiding cluttering your project everytime you build on Visual Studio.
+1. Create a vue.config.js file on ./ClientApp/ as below, for configuring webpack and setting the wwwroot folder:
+```js
+module.exports = {
+	outputDir: '../wwwroot',
+
+	configureWebpack: {
+		output: {
+			devtoolModuleFilenameTemplate: '[absolute-resource-path]',
+			devtoolFallbackModuleFilenameTemplate: '[absolute-resource-path]?[hash]'
+		}
+	}
+};
+```
+
+##### Building with vue-cli 3.0
+There is an unknown issue with the vue-cli 3.0 that prevents automating the build process. Everytime you wish to refresh the wwwroot folder you will need to run the command `npm run build` on the ClientApp folder.
+
+## Limitations
+
+* Lang attribute only supports JavaScript and TypeScript languages.
+* Lang attribute doesn't work with template or style tags.
+* Debugging .vue files is not supported due to its preprocessed nature.
```

Comment:
"Debugging script blocks in .vue files..."

Reply:
Fixed.